### PR TITLE
feat: added support for the edge switchover

### PR DESCRIPTION
**Pull request by Kapim**

### CODE REVIEW

File: era_5g_client/client.py [hunk 1]
```diff
@@ -183,13 +186,25 @@ def register(
 
         super().register(netapp_address, args, wait_until_available, wait_timeout)
 
+    def netapp_address_changed(self):
```
Comment (ZdenekM) [inline era_5g_client/client.py:189]: Add some docstring?

File: era_5g_client/client.py [hunk 2]
```diff
@@ -81,6 +81,8 @@ def __init__(
         self.resource_checker: Optional[MiddlewareResourceChecker] = None
         self.middleware_info: Optional[MiddlewareInfo] = None
         self.token: Optional[str] = None
+        self.args: Optional[Dict[str, Any]] = None
+        self.switching: bool = False
```
Comment (ZdenekM) [inline era_5g_client/client.py:85]: Maybe make this "readonly" property? I mean like:

```python
def __init__():
    self._switching = False

@property
def switching() -> bool:
    """ You can document this and it is 'read-only'. """
    return self._switching

def netapp_address_changed(self):
    self._switching = True
    ...
```

File: era_5g_client/client.py [hunk 3]
```diff
@@ -183,13 +186,25 @@ def register(
...
+        self.disconnect()
+        self.register(self.netapp_address, self.args, True)
+        self.switching = False
+
     def disconnect(self) -> None:
-        """Disconnects the WebSocket connection, stop resource checker and delete resources."""
+        """Disconnects the WebSocket connection, stop resource checker and delete resources.
+
+        Args:
+            finish (bool, optional): If true, the resource checker is finished and all resources are deleted from the server
```
Comment (ZdenekM) [inline era_5g_client/client.py:200]: outdated

File: era_5g_client/middleware_resource_checker.py [hunk 4]
```diff
@@ -15,15 +15,20 @@ class MiddlewareResourceChecker(Thread):
...
+        **kw,
     ) -> None:
         """Constructor.
 
         Args:
             token (str): Login token.
             action_plan_id (str): Action plan ID.
             status_endpoint (str): Status endpoint.
-            state_callback (Callable, optional): Optional state callback.
+            url_changed_callback (Callable, optional): Trigered if the recieved url has changed.
```
Comment (ZdenekM) [inline era_5g_client/middleware_resource_checker.py:31]: received

File: era_5g_client/middleware_resource_checker.py [hunk 5]
```diff
@@ -50,18 +55,19 @@ def run(self) -> None:
 
         while not self.stop_event.is_set():
             resource_state = self.get_resource_status()
-
+            print(resource_state)
```
Comment (ZdenekM) [inline era_5g_client/middleware_resource_checker.py:58]: remove

File: era_5g_client/middleware_resource_checker.py [hunk 6]
```diff
@@ -82,8 +88,9 @@ def get_resource_status(self) -> Dict:
             else:
                 logger.debug(e)
             raise FailedToConnect(f"Could not get the resource status, revisit the log files for more details. {e}")
-
+        print(response)
```
Comment (ZdenekM) [inline era_5g_client/middleware_resource_checker.py:91]: remove

File: era_5g_client/middleware_resource_checker.py [hunk 7]
```diff
@@ -82,8 +88,9 @@ def get_resource_status(self) -> Dict:
...
         resp = response.json()
+        print(resp)
```
Comment (ZdenekM) [inline era_5g_client/middleware_resource_checker.py:93]: remove

File: tests/dummy_middleware.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import uuid
+from typing import Tuple
+
+from flask import Flask, Response, jsonify
+
+app = Flask(__name__)
+
+
+url = "http://localhost:5896"
```
Comment (ZdenekM) [inline tests/dummy_middleware.py:9]: Just a note: if this is going to be used for tests, the port should be random, as tests could be running in parallel.

**Review by ZdenekM — CHANGES_REQUESTED** [review]
Also it would be nice to squash it into one commit.

File: era_5g_client/client.py (hunk 2 above)
Comment (Kapim) [reply era_5g_client/client.py:85]: ok, changed

File: era_5g_client/client.py (hunk 1 above)
Comment (Kapim) [reply era_5g_client/client.py:189]: added

File: era_5g_client/client.py (hunk 3 above)
Comment (Kapim) [reply era_5g_client/client.py:200]: removed

File: era_5g_client/middleware_resource_checker.py (hunk 4 above)
Comment (Kapim) [reply era_5g_client/middleware_resource_checker.py:31]: updated

File: tests/dummy_middleware.py (hunk 8 above)
Comment (Kapim) [reply tests/dummy_middleware.py]: At the moment, it is only meant for manual testing.

**Review by ZdenekM — APPROVED** [review]

**Pull request merged**